cloud/amazon/sqs_queue.py

```diff
@@ -113,16 +148,23 @@ def create_or_update_sqs_queue(connection, module):
     try:
         queue = connection.get_queue(queue_name)
         if queue:
-            # Update existing
+            # Update existing            
             result['changed'] = update_sqs_queue(queue, check_mode=module.check_mode, **queue_attributes)
-
         else:
             # Create new
             if not module.check_mode:
                 queue = connection.create_queue(queue_name)
                 update_sqs_queue(queue, **queue_attributes)
             result['changed'] = True
-
+        
+        if not module.check_mode:
```

Comment:
Yes, that would technically work.

It may look a bit strange though that the attributes have their original values instead of the updated ones.  To be accurate, you may need to load the original values, then override with the request values.

I'll leave it up to @emusgrave if he wants to include this or whether it should be a separate PR.